hibernate-core/src/main/java/org/hibernate/graph/EntityGraphs.java

```diff
@@ -0,0 +1,352 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.graph;
+
+import java.text.ParseException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.persistence.AttributeNode;
+import javax.persistence.EntityGraph;
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+import javax.persistence.Subgraph;
+import javax.persistence.TypedQuery;
+
+/**
+ * A collection of {@link EntityGraph} utilities.
+ * These methods really belong inside other classes that we cannot modify, hence here.
+ * 
+ * @author asusnjar
+ */
+public final class EntityGraphs {
+
+	public static String HINT_FETCHGRAPH = "javax.persistence.fetchgraph";
```

Comment:
These should be unified with `org.hibernate.jpa.QueryHints#HINT_FETCHGRAPH` and `#HINT_LOADGRAPH`

Reply:
I agree. I remember looking into this but I think only in JPA (i.e. not Hibernate specific). It would be ideal if those are defined there... but they are not.

Reply:
Not sure what you mean here

Reply:
What I mean by "I agree"? :) ... I agree with you.

Let me paraphrase:

1. When this was written the goal was to try to be as standard JPA as possible and not Hibernate specific. Unfortunately, I didn't find these constants anywhere but in org.hibernate.*. I may also have (don't remember now) hit some cyclic references in Hibernate itself - i.e. not having that constant available in my dependencies.

It would be ideal if these constants are defined in JP. Better yet, this entire class, EntityGraphs, should really have to exist. It enriches methods of other classes only, namely Query, EntityManager, EntityGraph, Subgraph and AttributeNode, all in javax.persistence.

Reply:
Edited some typos/similar above. The reason we went ahead with this class is because we feel that the spirit of the approach is correct and that something like this must eventually find its way into JP. As this adds nothing on top of what JP offers now (other than convenience), it should be easy/trivial to not only reimplement this class to use the then-available JP implementation but also to update any code using this class to use JP directly.